test/built-ins/Set/prototype/union/receiver-not-set.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2023 Kevin Gibbons, Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: Set.prototype.union throws when receiver is not a Set
+features: [Set-methods]
+---*/
+
+class MySetLike {
+  size = 2;
+  has = () => {};
+  keys = function* keys() {
```

Comment:
these tests probably shouldn't use class field syntax, since it's not necessary for the test